src/services/facilityService.ts

```diff
@@ -1,13 +1,48 @@
-import { facilities } from '../mockData/mockData';
 import { Facility } from '../typeDefs/gqlTypes';
+import {Facility as PrismaFacility, Contact as PrismaContact } from '@prisma/client';
+
+import prisma from '../db/client';
+
+type FacilityAndRelations = (PrismaFacility & {
+    contact: PrismaContact;
+})
+
+function convertPrismaToGraphQLFacility(input: FacilityAndRelations) {
+    return {
+        id: String(input.id),
+        nameJa: input.nameJa,
+        nameEn: input.nameEn,
+        contact: input.contact,
+        healthcareProfessionals: []
+    } as Facility;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getFacilityById = (id: string) => {
-    const matchingFacility = facilities.find(
-        (location: Facility) => location.id === id
+export const getFacilityById = async (id: string) => {
+    const found = await prisma.facility.findUnique(
+        {where: {
+            id: parseInt(id)
+        },
+        include: {
+            contact: true
+        }}
     );
 
-    return matchingFacility;
+    return found ? convertPrismaToGraphQLFacility(found) : null;
```

Comment:
nit optimization: you could move the null check into the function so you don't have to keep doing the null check before calling

Reply:
that's a good idea and now I'm wondering why i didn't think of it 🦞